lib/haml_lint/reporter/disabled_config_reporter.rb

```diff
@@ -0,0 +1,104 @@
+require 'haml_lint/reporter/progress_reporter'
+
+module HamlLint
+  # Outputs a YAML configuration file based on existing violations.
+  class Reporter::DisabledConfigReporter < Reporter::ProgressReporter
+    HEADING =
+      ['# This configuration was generated by',
+       '# `haml-lint --auto-gen-config`',
+       "# on #{Time.now} using Haml-Lint version #{HamlLint::VERSION}.",
+       '# The point is for the user to remove these configuration records',
+       '# one by one as the lints are removed from the code base.',
+       '# Note that changes in the inspected code, or installation of new',
+       '# versions of Haml-Lint, may require this file to be generated again.']
+      .join("\n")
+
+    # Disables this reporter on the CLI since it doesn't output anything.
+    #
+    # @return [false]
+    def self.available?
+      false
+    end
+
+    # Create the reporter that will display the report and write the config.
+    #
+    # @param _log [HamlLint::Logger]
+    def initialize(_log)
+      super
+      @linters_with_lints = Hash.new { |hash, key| hash[key] = [] }
+      @linters_lint_count = Hash.new(0)
+    end
+
+    # A hash of linters with the files that have that type of lint.
+    #
+    # @return [Hash<String, Integer] a Hash with linter name keys and lint
+    #   count values
+    attr_reader :linters_lint_count
+
+    # A hash of linters with the files that have that type of lint.
+    #
+    # @return [Hash<String, Array<String>>] a Hash with linter name keys and file
+    #   name list values
+    attr_reader :linters_with_lints
+
+    # Prints the standard progress reporter output and writes the new config file.
+    #
+    # @param report [HamlLint::Report]
+    # @return [void]
+    def display_report(report)
+      super
+
+      File.write(ConfigurationLoader::AUTO_GENERATED_FILE, config_file_contents)
+      log.log "Created #{ConfigurationLoader::AUTO_GENERATED_FILE}."
```

Comment:
It's confusing that `super` in the constructor silently passes along the arguments, which in the ProgressReporter makes `log` available. It's been a while since I coded in Ruby, and this always comes back to confuse me... No action required, just me thinking out loud. 😄